assets/js/components/ReportError.js

```diff
@@ -51,43 +52,81 @@ export default function ReportError( { moduleSlug, error } ) {
 
 	const dispatch = useDispatch();
 
-	let title = sprintf(
-		/* translators: %s: module name */
-		__( 'Data error in %s', 'google-site-kit' ),
-		module?.name
-	);
-	let message = error.message;
+	const errors = Array.isArray( error ) ? error : [ error ];
+	const uniqueErrors = uniqueBy( errors, ( e ) => e.code && e.message );
```

Comment:
This doesn't look quite right, as it's not testing the combined equality of both `code` and `error` (due to the way `&&` short circuits only a falsy value of `code` _or_ the value of `message` will be used as the unique value), and it's also not taking `reconnectURL` into account.

I think this would probably be the right approach:

```js
import uniqWith from 'lodash/uniqWith`;

// ...

const uniqueErrors = uniqWith( errors, ( errorA, errorB ) =>
  errorA.code === errorB.code &&
  errorA.message === errorB.message &&
  errorA.data?.reconnectURL === errorB.data?.reconnectURL
);
```

Reply:
Or (as discussed "offline") maybe something like this would be more accurate -

```js
const uniqueErrors = errors
  .map( e => ( { message: getMessage( e.message ), reconnectURL: e.data?.reconnectURL }) )
  .uniqWith( (errorA, errorB) =>
    errorA.message === errorB.message &&
    errorA.reconnectURL === errorB.reconnectURL
  );
```

Reply:
Thanks, @techanvil, yes, this approach is good except that passing the mapped errors array as the first argument to `uniqWith`:

```js
const uniqueErrors = uniqWith(
	errors.map( ( e ) => ( {
		message: getMessage( e ),
		reconnectURL: e.data?.reconnectURL,
	} ) ),
	( errorA, errorB ) =>
		errorA.message === errorB.message &&
		errorA.reconnectURL === errorB.reconnectURL
);
```

Reply:
Thanks @hussain-t, good spot!